src/test/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImplTest.java

```diff
@@ -0,0 +1,647 @@
+        achIncomeFileTransaction.setOpenItemReferences(openItemReferences);
+
+        transactionGuts.add(achIncomeFileTransaction);
+        achIncomeFileTransactionSet.setTransactionGuts(transactionGuts);
+        achIncomeFileTransactionSets.add(achIncomeFileTransactionSet);
+        achIncomeFileGroup.setTransactionSets(achIncomeFileTransactionSets);
+
+        AchIncomeFileGroupTrailer achIncomeFileGroupTrailer = new AchIncomeFileGroupTrailer();
+        achIncomeFileGroupTrailer.setGroupControlNumber("0");
+        achIncomeFileGroupTrailer.setTotalTransactionSets(1);
+        achIncomeFileGroup.setGroupTrailer(achIncomeFileGroupTrailer);
+        groups.add(achIncomeFileGroup);
+        achIncomeFile.setGroups(groups);
+
+        AchIncomeFileTrailer achIncomeFileTrailer = new AchIncomeFileTrailer();
+        achIncomeFileTrailer.setInterchangeControlNumber("000000000");
+        achIncomeFileTrailer.setTotalGroups(1);
+
+        achIncomeFile.setTrailer(achIncomeFileTrailer);
+        achIncomeFiles.add(achIncomeFile);
+        return achIncomeFiles;
+    }
+
+    private AchIncomeFileTransactionOpenItemReference setupOpenItemReference(String type, String invoiceNumber) {
+        AchIncomeFileTransactionOpenItemReference openItemReference = new AchIncomeFileTransactionOpenItemReference();
+        openItemReference.setInvoiceAmount(new KualiDecimal("12761.79"));
+        openItemReference.setNetAmount(new KualiDecimal("12761.79"));
+        openItemReference.setType(type);
+        openItemReference.setInvoiceNumber(invoiceNumber);
+        return openItemReference;
+    }
+
+    protected List<AchIncomeNote> setupNotes(String noteText) {
+        List<AchIncomeNote> notes = new ArrayList<>();
+        AchIncomeNote note = new AchIncomeNote();
+        note.setNoteText(noteText);
+        notes.add(note);
+        return notes;
+    }
+
+    private class TestableAdvanceDepositServiceImpl extends AdvanceDepositServiceImpl {
+
+        @Override
+        protected Integer getSourceAccountingLinePostingYear(SourceAccountingLine sourceAccountingLine) {
+            return new Integer("2016");
+        }
+
+        @Override
+        protected void setSourceAccountingLineAccountNumber(String account, SourceAccountingLine sourceAccountingLine) {
+            Field accountNumber;
+            try {
+                accountNumber = sourceAccountingLine.getClass().getSuperclass().getDeclaredField("accountNumber");
+            } catch (NoSuchFieldException e) {
+                throw new RuntimeException("huh, how did this happen?", e);
+            }
+            accountNumber.setAccessible(true);
+            try {
+                accountNumber.set(sourceAccountingLine, account);
+            } catch (IllegalAccessException e) {
+                throw new RuntimeException("huh, how did this happen?", e);
+            }
+        }
+    }
+
+    private class MockParameterService implements ParameterService {
```

Comment:
yeah, let me look into that; I think when I first wrote this test, I hadn't rebased to pull in the new MockParameterServiceImpl yet